File: src/web/magmac/app/lang/node/PlantUMLFooter.ts
import { MapNode } from "../../../../magmac/app/compile/node/MapNode";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { ExactRule } from "../../../../magmac/app/compile/rule/ExactRule";
import { TypeRule } from "../../../../magmac/app/compile/rule/TypeRule";
export class PlantUMLFooter {
	static createRule() : TypeRule {return new TypeRule( "footer", new ExactRule( "@enduml"));;}
	public serialize() : Node {return new MapNode( "footer");;}
}

File: packages/compiler-java/src/main/java/magmac/app/lang/node/PlantUMLFooter.java
package magmac.app.lang.node;

/**
 * Represents the closing {@code @enduml} tag of a PlantUML diagram.
 */

import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;
import magmac.app.compile.rule.ExactRule;
import magmac.app.compile.rule.TypeRule;

public class PlantUMLFooter implements PlantUMLRootSegment {
    static TypeRule createRule() {
        return new TypeRule("footer", new ExactRule("@enduml"));
    }

    @Override
    public Node serialize() {
        return new MapNode("footer");
    }
}
